design-editor/src/panel/property/attribute/attribute-element-image.js

```diff
@@ -173,11 +173,13 @@ class AttributeImage extends DressElement {
 			const reader = new FileReader();
 			reader.addEventListener('loadend', event => {
 				if (event.target.readyState === FileReader.DONE) {
+					const { basePath } = utils.checkGlobalContext('globalData');
 					const dir = utils.checkGlobalContext('globalData').fileUrl.replace(/[^/]+$/gi, '');
-					const writepath = (`${dir  }/images/${  file.name}`).replace(/\/+/gi, '/');
-					const readpath = (`images/${  file.name}`).replace(/\/+/gi, '/');
 
-					utils.checkGlobalContext('writeFile')(
+					const writepath = path.join(dir, 'images', file.name);
```

Comment:
In module path-utils, we have the function named `createProjectPath` which make exactly the same as you here. Please use it to follow the DRY rule.

Reply:
good point - done :)